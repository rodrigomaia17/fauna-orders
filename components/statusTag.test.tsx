import React from 'react';
import { render } from '@testing-library/react';
import StatusTag from './statusTag';
import '@testing-library/jest-dom/extend-expect';

test('print the status passed as param', () => {
  const { getByText, container } = render(<StatusTag status="status" />);
  expect(getByText('status')).toBeVisible();
  expect(container.firstChild).toMatchSnapshot();
});

describe('Uses the correct color class', () => {
  const mapping = {
    processing: 'warning',
    approved: 'success',
    anyOther: 'danger'
  };

  for (const [key, value] of Object.entries(mapping)) {
    const { container } = render(<StatusTag status={key} />);
    const statusElement = container.firstChild as HTMLElement;
    const className = statusElement.className;
    expect(className).toContain('is-' + value);
  }
});
