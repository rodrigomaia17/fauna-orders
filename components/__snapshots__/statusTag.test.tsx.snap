// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`print the status passed as param 1`] = `
<span
  class="jsx-2445842882 box-line-status is-size-7-mobile tag is-danger"
>
  status
</span>
`;
